app/signer/bls.go

```diff
@@ -251,51 +254,124 @@ func (k *BlsKey) BlsPubKey() (bls12381.PublicKey, error) {
 	return k.PubKey, nil
 }
 
+// GetBlsKeyPassword is a unified function to handle BLS key password retrieval.
+// Returns password if a password was found from one of the sources.
+// 1. If noPassword is true, returns empty string without checking other sources.
+// 2. Environment variable BABYLON_BLS_PASSWORD
+// 3. Password file (ONLY if path is explicitly provided and file exists)
+// 4. Interactive prompt, if no other sources are found
+//   - For new keys: confirmation flow with matching passwords
+//   - For existing keys: single prompt without confirmation
+func GetBlsKeyPassword(noPassword bool, passwordFilePath string, isNewKey bool) (string, error) {
+	if err := ValidatePasswordMethods(noPassword, passwordFilePath); err != nil {
+		return "", err
+	}
+
+	// If using no-password mode, return empty password immediately
+	if noPassword {
+		return "", nil
+	}
+
+	// Try getting password from environment
+	password, found := GetBlsPasswordFromEnv()
+	if found && password != "" {
+		return password, nil
+	}
+
+	passwordBytes, err := os.ReadFile(passwordFilePath)
+	if err == nil {
+		return string(passwordBytes), nil
+	}
+
+	// Use appropriate interactive prompt based on whether this is a new key or existing key
+	if isNewKey {
+		return NewBlsPassword(), nil
+	}
+	return GetBlsUnlockPasswordFromPrompt(), nil
+}
+
+// ValidatePasswordMethods checks if multiple password methods are being used simultaneously.
+// Returns an error if more than one method is provided.
+func ValidatePasswordMethods(noPassword bool, passwordFilePath string) error {
+	password, found := GetBlsPasswordFromEnv()
+	envSet := password != "" && found
```

Comment:
Shouldn't we allow passing  `""` thorough env variable ? (this is main reason to use `LookUpEnv`)